rmgpy/molecule/resonance.py

```diff
@@ -1116,3 +1133,183 @@ def _clar_transformation(mol, aromatic_ring):
+                        structure.update_atomtypes(log_species=False)
+                    except AtomTypeError:
+                        pass
+                    else:
+                        structures.append(structure)
+    return structures
+
+def generate_adsorbate_double_shift_down_resonance_structures(mol):
+    """
+    Generate all of the resonance structures formed by the shift of four electrons from a C#C bond to increase the bond
+    order between two X-C atoms by 2.
+    Example XCXC: [X]C#C[X] <=> [X]#CC#[X]
+    (where '#' denotes a triple bond)
+    """
+    cython.declare(structures=list, paths=list, index=cython.int, structure=Graph)
+    cython.declare(atom=Vertex, atom1=Vertex, atom2=Vertex, atom3=Vertex, atom4=Vertex, bond12=Edge, bond23=Edge, bond34=Edge)
+    cython.declare(v1=Vertex, v2=Vertex)
+
+    structures = []
+    if mol.is_bidentate():
+        for atom in mol.vertices:
+            paths = pathfinder.find_adsorbate_delocalization_paths(atom)
+            for atom1, atom2, atom3, atom4, bond12, bond23, bond34 in paths:
+                if bond23.is_triple() and bond12.is_single() and bond34.is_single():
+                    bond12.increment_order()
+                    bond12.increment_order()
+                    bond23.decrement_order()
+                    bond23.decrement_order()
+                    bond34.increment_order()
+                    bond34.increment_order()
+                    structure = mol.copy(deep=True)
+                    bond12.decrement_order()
+                    bond12.decrement_order()
+                    bond23.increment_order()
+                    bond23.increment_order()
+                    bond34.decrement_order()
+                    bond34.decrement_order()
+                    try:
+                        structure.update_atomtypes(log_species=False)
+                    except AtomTypeError:
+                        pass
+                    else:
+                        structures.append(structure)
+    return structures
+
+def generate_adsorbate_conjugate_resonance_structures(mol):
+    """
+    Generate all of the resonance structures formed by the shift of two electrons in a conjugated pi bond system of
+    a bidentate adsorbate with a bridging C atom in between.
+    Example XCHCHXC: [X]#CC=C[X] <=> [X]=C=CC=[X]
+    (where '#' denotes a triple bond, '=' denotes a double bond)
+    """
+    cython.declare(structures=list, paths=list, index=cython.int, structure=Graph)
+    cython.declare(atom=Vertex, atom1=Vertex, atom2=Vertex, atom3=Vertex, atom4=Vertex, atom5=Vertex, bond12=Edge, bond23=Edge, bond34=Edge, bond45=Edge)
+    cython.declare(v1=Vertex, v2=Vertex)
+
+    structures = []
+    if mol.is_bidentate():
+        for atom in mol.vertices:
+            paths = pathfinder.find_adsorbate_conjugate_delocalization_paths(atom)
+            for atom1, atom2, atom3, atom4, atom45, bond12, bond23, bond34, bond45 in paths:
+                if bond23.is_single():
```

Comment:
Do you think this pathway could also be valid if `bond_23` is double?
If I apply the recipe for
`X1 {D/T} C2 {S} C3 {D/T} C4 {S/D} X5`
I get
`X1 {S/D} C2 {D} C3 {S/D} C4 {D/T} X5`
reversing the atom labels this is:
`X1 {D/T} C2 {S/D} C3 {D} C4 {S/D} X5`
I expected that the 3rd line will be identical to the 1st line, since the recipe could be equally applied for it as well.

So maybe we should check for these bonds to get a mirror image:
`X1 {D/T} C2 {S/D} C3 {D/T} C4 {S/D} X5`

Reply:
Thanks, Alon. You are right, `bond_23` could also be a double bond. I changed the if statement.